### PR TITLE
Edit notes command

### LINKED ISSUE #183: Add edit notes via CLI
Works by specifying which note index to edit via a new command editNotes CLI first.
There will be a separate edit GUI in the future.
Sort by alphabetical order will be removed in this task so index is kept consistent for the user(by date added).

### PR DESCRIPTION
**Pull request by sheenkerr**
Closes #183 & #224

Used editnotes as command word to follow the addnotes syntax.

### CODE REVIEW

File: src/main/java/seedu/address/logic/commands/EditNotesCommand.java
Comment (Nicholas-Cheng-De-Fei) [inline src/main/java/seedu/address/logic/commands/EditNotesCommand.java:1]: One bug I found was that the `ContactDetails` panel does not get updated after I edited the notes of a person.

File: src/main/java/seedu/address/logic/parser/AddressBookParser.java [hunk 1]
```diff
@@ -85,6 +86,8 @@ public Command parseCommand(String userInput) throws ParseException {
 
         case AddNotesCommand.COMMAND_WORD:
             return new AddNotesCommandParser().parse(arguments);
+        case EditNotesCommand.COMMAND_WORD:
```
Comment (Nicholas-Cheng-De-Fei) [inline src/main/java/seedu/address/logic/parser/AddressBookParser.java:89]: Add a line spacing above

**Review by Nicholas-Cheng-De-Fei — CHANGES_REQUESTED** [review]

**Review by Nicholas-Cheng-De-Fei — CHANGES_REQUESTED** [review]

File: src/main/java/seedu/address/logic/commands/EditNotesCommand.java
Comment (Nicholas-Cheng-De-Fei) [reply src/main/java/seedu/address/logic/commands/EditNotesCommand.java:1]: Move bug fix to v1.5

**Review by Nicholas-Cheng-De-Fei — APPROVED** [review]
LGTM

File: src/test/java/seedu/address/testutil/TypicalNotes.java [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+
+import seedu.address.model.person.Note;
+
+/**
+ * A utility class containing a list of {@code Notes} objects to be used in tests.
+ */
+public class TypicalNotes {
+    public static final Note TYPICAL_NOTE = new Note("Loves Ramen");
+    public static final Note DUPLICATE_NOTE = new Note("Likes dumplings");
+
```
Comment (ITLimJiaWei) [inline src/test/java/seedu/address/testutil/TypicalNotes.java:11]: Small formatting issue of blank space here

File: src/test/java/seedu/address/logic/commands/EditNotesCommandTest.java [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+    private Model model = new ModelManager(getTypicalAddressBook(), new UserPrefs());
+
+    @Test
+    public void execute_editNotesCommand_success() throws Exception {
+        // Create the person and expected models as in the original
+        Person personToEdit = model.getFilteredPersonList().get(INDEX_FIRST_PERSON.getZeroBased());
+
+        // Define the edited person with updated notes
+        Person editedPerson = new PersonBuilder(personToEdit).withNotes(
+                TYPICAL_NOTE.getNote(), "Likes dumplings").build();
```
Comment (ITLimJiaWei) [inline src/test/java/seedu/address/logic/commands/EditNotesCommandTest.java:43]: Better to have "Likes dumplings" defined as a final static var elsewhere but minor issue.

File: src/test/java/seedu/address/testutil/TypicalIndexes.java [hunk 4]
```diff
@@ -10,4 +10,8 @@ public class TypicalIndexes {
     public static final Index INDEX_SECOND_PERSON = Index.fromOneBased(2);
     public static final Index INDEX_THIRD_PERSON = Index.fromOneBased(3);
     public static final Index INDEX_OUTOFBOUND_PERSON = Index.fromOneBased(1000);
+
+
```
Comment (ITLimJiaWei) [inline src/test/java/seedu/address/testutil/TypicalIndexes.java:14]: Extra empty space here

**Review by ITLimJiaWei — APPROVED** [review]
Very minor formatting issues but overall LGTM

**Pull request merged**